### ISSUE TITLE
Drag items in nested ng repeat filtered

### ISSUE THREAD

**Issue by GDuchemin** [luckylooke/dragular#84]
Hi luckylooke !

Here is a JSfiddle to test nested ng-repeat filtered: https://jsfiddle.net/hLsuege1/
As you see and as your documentation, there is no dynamic model for `containersFilteredModel` option of dragularInstance.

I don't know how to have dynamic model for `containersFilteredModel` (obviously based on the number of column). Number of column is variable because i can add / remove column. Same for rows.

Every way to do that will be helpful ;) 

Thank you !

**Comment by blackendstudios** [comment]
yeah, I having the same issue, for example when using nested arrays if you try to sort them with filters in the html, it works just as it looks but when you start dragging the sorted nested array, it doesn't drag properly you got unexpected results, so by altering in this case 'sort', it has to be directly on the source array right?  not by using filters

**Comment by GDuchemin** [comment]
@blackendstudios I tried to filter the collection in controller and push it in an array for model, i doesn't work. Even if I kill dragular and instantiate it again with this filtered array, i have a strange bug which allow user to drop items in their container. I think the model and filtered model is not clean by this way but I really don't know why.

The only way I currently find to patch this issue its to reload my route (not my location) at each drop, to keep filtered array, pending on this feature ;)

**Comment by blackendstudios** [comment]
@GDuchemin but by reloading the state isn't too much 'overhead' in order to satisfy our app requirements? maybe 'dragularService' can have the functionality of 'containersFilteredModel' being configurable like set the amount of models by referring then by the index of each nested array...or I don't know jajaja I just can't imagine that

**Comment by GDuchemin** [comment]
I don't know... I've tried anything ! the functionality of 'containersFilteredModel' has arrays contain filtered model, and i can clean them and push again on them. I'm waiting for a fix doc and maybe a fix feature.

**Comment by luckylooke** [comment]
Hi guys, it seems that nested repeats with filtering is getting to be viral, I hope I will have some time soon to have look at it. Thanks for patience.

**Comment by GDuchemin** [comment]
Thanks for your help lucky.

**Comment by amonsot890** [comment]
hi good day @luckylooke , i'm currently using dragular for our app, but it seems that using it with nested ng-repaeat is a problem, can you help me solve this problem . :) thank you.

**Comment by luckylooke** [comment]
@amonsot890 nested repead is not problem since you are not filtering the parent repeat see [example](http://luckylooke.github.io/dragular/#/docs/exampleNestedNgRepeatWithModel). Filtered parent is problem for now, I dont have much time for solving it right now :/

**Comment by amonsot890** [comment]
@luckylooke  thanks for the reply but i forgot to mention that i'm doing it with filters, 

here is my html code, 

<div class="_column" ng-repeat="status in fi.get.kanban.active.status.status">
                                <div id="head{{::$index}}" class="_header">
                                    <label>{{status}}</label>
                                </div>
                                <div class="_body">
                                    <div slim-scroll="mc.scrollOption">
                                        
                                        <div class='wrapper'>
                                            <div class='tableRow'>
                                                <div class='_drag' dragular="dragularOption()" id="{{::$index}}" style="min-height: 100px;">
                                                    <div id="{{issue.key}}" key="{{issue._id}}" class=" _repeat" ng-include="'/_template/task.kanban.client.view.html'"
                                                         ng-repeat="issue in fi.issues[$index] track by $index" ng-click="def.viewIssue(issue.key)">
                                                    </div>
                                                </div>
                                            </div>
                                        </div>
                                    </div>
                                </div>
                            </div>
i wish i could show you in person the whole code structure that i am working right now. Thanks a lot..:)

**Comment by luckylooke** [comment]
I am looking into your code and I cant see nested ng-repeat, could you please raise another issue for your problem as it is not relevant to this issue. Thanks

**Comment by GDuchemin** [comment]
@luckylooke, did you fix something or no-filtered parents worked before ?

**Comment by luckylooke** [comment]
@GDuchemin I didnt move with this issue, but non-filtered parents ng-repeat should work. Filtered ng-repeat doesn't support child ng-repeats as it lost connection with true model, that is the problem from point of implementation. Because filtered repeats works with cloned model.

**Comment by GDuchemin** [comment]
Look at my fiddle on first post, i just filtered child ng-repeat and it's not working. To test, just try to delete by clicking in (X). Bad item is deleted.

**Comment by luckylooke** [comment]
Ok.. I forgot.. so it seems to be not working too.

Dňa 17.10.2016 15:43 používateľ "Guillaume Duchemin" <
notifications@github.com> napísal:

**Comment by GDuchemin** [comment]
Do you think that you can fix it soon ?

**Comment by luckylooke** [comment]
It depends, what is soon for you :) I can only guess.. :/

**Comment by amonsot890** [comment]
@luckylooke thank you very much.. great work by the way.

**Comment by luckylooke** [comment]
@GDuchemin Can you check if this update of your fiddle is what you expected? https://jsfiddle.net/rjLoyagm/2/ Thanks

**Comment by luckylooke** [comment]
I was just looking the source code of library, and I didn't found the reason why nested filtered ng-repeats shouldn't work. It seems that it is just very hard to implement. So I will create such example with detailed comments soon. Maybe I bring more light into the problem. Thanks for patience.

**Comment by luckylooke** [comment]
@amonsot890 please try to reproduce your problem in fiddle, maybe I can help you with your problem too ;)

**Comment by amonsot890** [comment]
@luckylooke  sorry for the late response, hmm ill try.  and one more thing i have the same problem with this example https://jsfiddle.net/rjLoyagm/2/ , but when i try to remove the containersModel option, it worked fine  on my side, but when i tried it with websockets and tried emiting with a group of people. it seems that it cant contain the whole thing, i'am trying to configure some of the works but still can't solve the problem. hmmm.

**Comment by amonsot890** [comment]
@luckylooke  thanks, by the way for your effort and help. really appreciate it.

**Comment by luckylooke** [comment]
@amonsot890 I am so soory, but I am lost in your comment. Could you please send me step by step for reproducing the problem from start to the point where app doesnt act as expected. Something like (example):
1. I click on button "some button"
2. I drag from item "itemName" from 2nd column to "itemIn3rdColName" place in 3rd column
expecting to "itemName" disapear from 2nd column, and show abbove "itemIn3rdColName" in 3rd, BUT "itemName" is still in 2nd column.

Please try to give me such instructions to reproduce the problem, like for dummies :D It would be very important for solving the problem. Thank you very much for your patience ;)
